### ISSUE TITLE
`weave setup` fails with "docker: not found"

### ISSUE THREAD

**Issue by rade** [weaveworks/weave#3121]
The latest snapshot fails in `weave setup`:
```
$ weave setup
latest: Pulling from weaveworks/weaveexec
90f4dba627d6: Already exists 
c3a059070f8c: Already exists 
be7f708e3d16: Already exists 
c6f0654dab6f: Already exists 
bc0afda9d37e: Already exists 
4ef170add6a4: Already exists 
4d11e653a953: Already exists 
d4243b49cf87: Already exists 
3aaf76f66a27: Already exists 
d567fa3b867a: Already exists 
Digest: sha256:17f92dc2b14535d6e99bb040ede5c816076e1df1ce0c96f53a8120bf864c9c3a
Status: Image is up to date for weaveworks/weaveexec:latest
Using default tag: latest
latest: Pulling from weaveworks/weavedb
Digest: sha256:98ddf536104922d1484571de2cd0464db2d00e8c0b7d186dca331a41f0f77b06
Status: Image is up to date for weaveworks/weavedb:latest
/home/weave/weave: line 246: docker: not found
```

The problem here is that `setup` has two halves, which need to execute at the client and server (inside weaveexec), respectively, but is missing a `[ -z "$IS_LOCAL" ]` guard, just like there is on the `version` branch. Hence the client part is executed at the server too, which is superfluous and fails since the docker exe was removed from weaveexec.

cc @dtshepherd

**Comment by rade** [comment]
We really should add a `weave setup` test.

**Comment by dtshepherd** [comment]
Ugh, I know I tested the `docker pull` change manually, but I'm guessing I did that test prior to removing the `docker` tgz logic in the Makefile.

I started a bugfix branch and commited this change: https://github.com/weaveworks/weave/commit/bc7dc7b71a9c424779a971761b78551a2f62c7ae

But I'm having issues testing it.  The `docker pull` locally actually breaks the test because it pulls the `weaveexec:latest` from Docker Hub which doens't have the fix!  I'll try to look at it more this weekend unless someone else has some ideas...

**Comment by dtshepherd** [comment]
@rade @bboreham any thoughts on how to avoid the issue with `docker pull` overwriting the images that were just built when trying to test `weave setup`?  I could create some kind of testing flag to skip `docker pull` locally or alias it to be a no-op locally when testing (or set some test file or env variable that I can checked in the test).

**Comment by bboreham** [comment]
One way to test `weave setup` would be to run a local registry which the build is pushed to, first.
Or push it to dockerhub under the git tag?
I'm not sure I see that work as worthwhile.

**Comment by dtshepherd** [comment]
Yeah both of those seem painful. Maybe just push the fix without the test for now?

**Comment by bboreham** [comment]
Done - b8d7af3822c0e37cef0b6cab8732e34c95f30b1f

**Comment by rade** [comment]
That guard should really go outside the `for` loop.

**Comment by rade** [comment]
> That guard should really go outside the `for` loop.

Fixed in 89c7e6d33
